### PR TITLE
fix: fix GLIBC version not found in java maven jar

### PR DESCRIPTION
**Pull request by LuQQiu**
Revert nodejs version back to 16 to avoid GLIBC version not found issue in ubuntu and other OS.
Starting from nodejs 18, it requires a much higher GLIBC version and will prevent the application from starting if fail the check.

### CODE REVIEW

**Comment by LuQQiu** [comment]
@eddyxu PTAL, i cancelled the workflow since it will actually release to maven.
If the changes look good, i will update the maven version to 0.0.5 and let the pipeline runs

**Review by eddyxu — COMMENTED** [review]
Can we make a 0.0.5 test release to see whether it solve the problem

**Comment by LuQQiu** [comment]
```
Run echo "use-agent" >> ~/.gnupg/gpg.conf
/home/runner/work/_temp/e0369203-8970-4980-8b5c-b283201cd97a.sh: line 1: /home/runner/.gnupg/gpg.conf: No such file or directory
Error: Process completed with exit code 1.
```

The setup-java didn't set up gpg correctly
```
Run actions/setup-java@v4
  with:
    distribution: temurin
    java-version: 8
    cache: maven
    server-id: ossrh
    server-username: SONATYPE_USER
    server-password: SONATYPE_TOKEN
    java-package: jdk
    check-latest: false
    overwrite-settings: true
    job-status: success
    token: ***
```

The previous successful one shows
```
Run actions/setup-java@v4
  with:
    distribution: temurin
    java-version: 8
    cache: maven
    server-id: ossrh
    server-username: SONATYPE_USER
    server-password: SONATYPE_TOKEN
    gpg-private-key: ***
    gpg-passphrase: ***
    java-package: jdk
    check-latest: false
    overwrite-settings: true
    job-status: success
    token: ***
```

File: .github/workflows/java-publish.yml [hunk 1]
```diff
@@ -40,6 +44,10 @@ jobs:
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
+      - name: Set up Node.js 16
```
Comment (walterddr) [inline .github/workflows/java-publish.yml:47]: apparently this fixes the GLIBC version issue?

Comment (LuQQiu) [reply .github/workflows/java-publish.yml:47]: This fixes the glibc issue in older ubuntu version, ubuntu version <= 20

**Pull request closed without merging**